mapper/mapper.go

```diff
@@ -709,6 +710,49 @@ func CreateTimePage(req *http.Request, f filter.Model, d dataset.DatasetDetails,
 		p.Data.SelectedEndMonth = selDates[len(selDates)-1].Month().String()
 		p.Data.SelectedEndYear = fmt.Sprintf("%d", selDates[len(selDates)-1].Year())
 	}
+	var minYear, maxYear string
+	var selectedMonths []string
+	for _, selVal := range selVals {
+		month, err := time.Parse("Jan-06", selVal.Option)
+		if err != nil {
+			log.Event(ctx, "unable to convert date to month value", log.ERROR, log.Error(err))
+			continue
+		}
+		monthStr := month.Format("January")
+		if !fdHelpers.StringInSlice(monthStr, selectedMonths) {
+			selectedMonths = append(selectedMonths, monthStr)
+		}
+		yearStr := month.Format("2006")
+		if minYear == "" {
+			minYear = yearStr
+		}
+		if maxYear == "" {
+			maxYear = yearStr
+		}
+		yearInt, err := strconv.Atoi(yearStr)
+		maxYearInt, err := strconv.Atoi(maxYear)
+		minYearInt, err := strconv.Atoi(minYear)
+		if yearInt > maxYearInt {
+			maxYear = yearStr
+		} else if yearInt < minYearInt {
+			minYear = yearStr
+		}
+	}
+	var listOfAllMonths []timeModel.Month
+	for i := 0; i < 12; i++ {
```

Comment:
**Magic number** ⚠️  making `12` a constant with a sensible name will make it crystal clear what this is. I know it seems obvious but its best practice to avoid magic numbers in code.